docs/optimization_models.rst

```diff
@@ -2,4 +2,619 @@
+* :ref:`Results`
+
+.. _Overview:
+
+Overview
+~~~~~~~~
+
+The optimization of a given *ThermalNetwork* is executed by:
+
+.. code-block:: python
+
+    import dhnx
+
+    tnw = dhnx.network.ThermalNetwork()
+    tnw = network.from_csv_folder('path/to/thermal_network')
+
+    invest_opt = dhnx.input_output.load_invest_options('path/to/invest_options')
+
+    settgs = {
+        'keyword1': value1,
+        'keyword2': value2,
+    }
+
+    tnw.optimize_investment(settings=settgs, invest_options=invest_opt)
+
+For executing an optimization, you must provide the optimization `settings`
+and the `invest_options` additional to the previous data, which defines a
+*ThermalNetwork*. Both are explained in the following section.
+
+.. _Input Data:
+
+Input Data
+~~~~~~~~~~
+
+In this section, it is firstly revised, what input data is exactly necessary
+from the *ThemalNetwork* class, and then explained, what data needs to be
+provided as `invest_options` and as `settings`.
+
+The following figure provides an overview of the input data:
+
+.. 	figure:: _static/optimization_input_data.svg
+   :width: 100 %
+   :alt: optimization_input_data.svg
+   :align: center
+
+   Fig. 1: Optimization Input Data
+
+The structure of the input data might look a bit confusing at the beginning, but provides a lot of
+options for building up complex district heating models. There are two groups of data:
+geo-referenced data and parameters for the investment optimisation (with might include other
+oemof-solph related data). All data needs to be provided in csv files. This means the geo-referenced
+data does not need to be geo-referenced for this model at all, but probably in many cases, it is the
+export of four geo-referenced layers (e.g. geopandasdataframe, shp-file, or any other), which are
+a line layer representing the potential places for the DHS-trenches, and three point layers for the
+producers, the consumers, and the potential forks of the DHS system. All geometry information of
+the network system is passed by an *id* for each element. Thus, the line layer connects all points
+and provides the spatial relation with the attributes *from_node*, *to_node*, and *length*. If you
+prepare the data, be careful that every consumer is connected to an edge/line, and every piping
+network system is connected to at least one producer.
+
+.. _TN_Input :
+
+ThermalNetwork
+""""""""""""""
+
+The data for the *ThermalNetwork* must be provided in the structure as defined
+for the .csv reader. The following data is required for applying an
+optimization:
+
+.. code-block::
+
+    tree
+    ├── pipes.csv                       # (required)
+    ├── consumers.csv                   # (required)
+    ├── forks.csv                       # (required)
+    ├── producers.csv                   # (required)
+    └── sequences                       # (optional)
+        └── consumers-heat_flow.csv
+
+The attributes, which are required, and which are optional with respect
```

Comment:
It is very helpful to know which attributes are necessary in particular for optimisation. Good to have these tables. As an idea for a cleaner structure, we could in a future release define all requirements in the respective specificatoins of dhnx/components_attrs. I will open an issue.

Reply:
The issue is #52. We can do this later.

Reply:
Yes, I think this would be very helpful in future. Thanks for opening the issue.